src/openttd.cpp

```diff
@@ -1092,6 +1092,18 @@ void SwitchToMode(SwitchMode new_mode)
 			break;
 
 		case SM_RESTARTGAME: // Restart --> 'Random game' with current settings
+			if (_game_mode == GM_EDITOR && _file_to_saveload.abstract_ftype == FT_SAVEGAME || _file_to_saveload.abstract_ftype == FT_SCENARIO) {
+				_game_mode == GM_EDITOR ? _switch_mode = SM_LOAD_SCENARIO : _switch_mode = SM_LOAD_GAME;
+				SwitchToMode(_switch_mode);
+				break;
+			}
+
+			if (_file_to_saveload.abstract_ftype == FT_HEIGHTMAP) {
```

Comment:
maybe make this an else if with a single break at the end?

Reply:
Done